app/src/main/java/com/duckduckgo/app/browser/BrowserTabFragment.kt

```diff
@@ -424,6 +448,33 @@ class BrowserTabFragment : Fragment(), FindListener, CoroutineScope {
             is Command.LaunchLegacyAddWidget -> launchLegacyAddWidget()
             is Command.RequiresAuthentication -> showAuthenticationDialog(it.request)
             is Command.SaveCredentials -> saveBasicAuthCredentials(it.request, it.credentials)
+            is Command.GenerateWebViewPreviewImage -> generateWebViewPreviewImage(it.forceImmediate)
+            is Command.LaunchTabSwitcher -> launchTabSwitcher()
+            is Command.LaunchTabSwitcherLegacy -> launchTabSwitcherLegacy()
+        }
+    }
+
+    private fun generateWebViewPreviewImage(forceImmediate: Boolean) {
+        webView?.let { webView ->
+
+            // if there's an existing job for generating a preview, cancel that in favor of the new request
+            bitmapGeneratorJob?.cancel()
```

Comment:
Optional: Should we do the opposite here and only run the job if it isn't already in progress (unless force immediate is true)? I'm assuming that if the job is requested multiple times in quick succession then the previews will be pretty similar. Feel free to ignore if that's not the case!

Reply:
I figure that we might as well favour getting the latest image available if called multiple times rather than favouring the original request which has _slightly_ more chance of being invalid.